### DISCUSSION TITLE
Inline shell script in bat/config?

### DISCUSSION THREAD

**Discussion by tony-sol** [sharkdp/bat#2785] [Q&A] [answered]
I've tried to make bat switch theme depends on current system style (light or dark), like
```
--theme "Visual Studio $(defaults read -g AppleInterfaceStyle || echo 'Light')+"
```
And it works ok for provided argument, but works while it defined in bat/config file:
```
[bat warning]: Unknown theme 'Visual Studio $(defaults read -g AppleInterfaceStyle 2>/dev/null || echo 'Light')+', using default.
```

Is there a way to allow inline shell script in bat configs?

**Comment by keith-hall** [comment]
This is currently not possible unfortunately. But it's an interesting use case. The best alternative I can think of for now would be to have a shell PROMPT which re-evaluates the `BAT_THEME` env var on each invocation...

**Reply by tony-sol** [reply, marked as answer]
Yeah, i did something similar, like

```shell
export __BAT_THEME="\$(defaults read -globalDomain AppleInterfaceStyle &> /dev/null && echo vs-dark || echo vs-light)"

alias cat="bat --paging=never ${__BAT_THEME:+--theme=\"$__BAT_THEME\"}"
```

Works pretty well
